### PR TITLE
cleanup: use i18n.T on all command descriptions

### LINKED ISSUE #910: Ensure we're using i18n where possible
There is a scattered usage of `i18n.T()` throughout descriptions and flags. Let's make future translations easier by ensuring we're using i18n where feasible.

Examples:

https://github.com/kubernetes/kubectl/blob/19b50d838ed2e2a7389a3fb467b259982c772b73/pkg/cmd/apply/apply.go#L171
https://github.com/kubernetes/kubectl/blob/19b50d838ed2e2a7389a3fb467b259982c772b73/pkg/cmd/annotate/annotate.go#L153

/sig cli
/kind cleanup
/priority backlog
/good-first-issue

### PR DESCRIPTION
**Pull request by HerrNaN**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
There is a scattered usage of `i18n.T()` throughout descriptions and flags. This make future translations easier by ensuring we're using `i18n` where feasible.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubectl/issues/910

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
applies translations on all command descriptions
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig cli

### CODE REVIEW

**Comment by HerrNaN** [comment]
The Linux Foundation seems to be migrating at the moment so I'll try to sign the agreement later.

Regarding the release notes I'm not quite sure if one is required (but I added one since the changes affect what the user sees) or what it should say.

**Comment by HerrNaN** [comment]
The agreement is signed now :+1:

File: staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ package apiresources
 import (
 	"fmt"
 	"io"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go:22]: Please organize imports: stdlib, 3rd party, k8s.

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ package apply
 
 import (
 	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go:21]: Ditto

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go [hunk 3]
```diff
@@ -62,8 +63,8 @@ func NewCmdApplyEditLastApplied(f cmdutil.Factory, ioStreams genericclioptions.I
 	cmd := &cobra.Command{
 		Use:                   "edit-last-applied (RESOURCE/NAME | -f FILENAME)",
 		DisableFlagsInUseLine: true,
-		Short:                 "Edit latest last-applied-configuration annotations of a resource/object",
-		Long:                  applyEditLastAppliedLong,
+		Short:                 i18n.T("Edit latest last-applied-configuration annotations of a resource/object"),
+		Long:                  i18n.T(applyEditLastAppliedLong),
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go:67]: `i18n.T` should be invoked on a text, not on a variable, like everywhere else

File: staging/src/k8s.io/kubectl/pkg/cmd/get/get.go [hunk 4]
```diff
@@ -159,7 +159,7 @@ func NewCmdGet(parent string, f cmdutil.Factory, streams genericclioptions.IOStr
 		Use:                   "get [(-o|--output=)json|yaml|wide|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=...] (TYPE[.VERSION][.GROUP] [NAME | -l label] | TYPE[.VERSION][.GROUP]/NAME ...) [flags]",
 		DisableFlagsInUseLine: true,
 		Short:                 i18n.T("Display one or many resources"),
-		Long:                  getLong + "\n\n" + cmdutil.SuggestAPIResources(parent),
+		Long:                  i18n.T(getLong + "\n\n" + cmdutil.SuggestAPIResources(parent)),
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/get/get.go:162]: on `getLong` not here

File: staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go [hunk 5]
```diff
@@ -19,6 +19,7 @@ package set
 import (
 	"errors"
 	"fmt"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go:22]: Import order

File: staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go [hunk 6]
```diff
@@ -21,6 +21,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go:24]: Import order

**Review by soltysh — CHANGES_REQUESTED** [review]
/ok-to-test

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go (hunk 3 above)
Comment (HerrNaN) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go:67]: @soltysh If there is a long text that have variables concatinated inside it, should I wrap the entire text with the translation or only wrap the string literal parts?

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go:67]: From what I see there are no variables being injected, and if they where I'm pretty sure that they should not be translated.

Comment (HerrNaN) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go:67]: @soltysh So do I understand you correctly when I say that `i18n.T()` should only ever be applied to string literals, and nothing else?

Comment (HerrNaN) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_edit_last_applied.go]: @soltysh Yes I understand that but in [this example](https://github.com/kubernetes/kubernetes/blob/56833a63c902e8d350d7335ef91746f554def967/staging/src/k8s.io/kubectl/pkg/cmd/config/config.go#L43-L50) there are variables intertwined with the text, should the `i18n.T()` wrap the entire thing or only the string literal parts?

File: staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go [hunk 7]
```diff
@@ -38,15 +38,16 @@ import (
 	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	"k8s.io/kubectl/pkg/polymorphichelpers"
 	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/i18n"
 	"k8s.io/kubectl/pkg/util/templates"
 )
 
 var (
 	validEnvNameRegexp = regexp.MustCompile("[^a-zA-Z0-9_]")
```
Comment (ramnar) [inline staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go:46]: pod (po), replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs)  is language independent. I believe this doesnt need usage of i18n.T()

File: staging/src/k8s.io/kubectl/pkg/cmd/config/rename_context.go [hunk 8]
```diff
@@ -63,7 +64,7 @@ func NewCmdConfigRenameContext(out io.Writer, configAccess clientcmd.ConfigAcces
 	cmd := &cobra.Command{
 		Use:                   renameContextUse,
 		DisableFlagsInUseLine: true,
-		Short:                 renameContextShort,
+		Short:                 i18n.T(renameContextShort),
```
Comment (ramnar) [inline staging/src/k8s.io/kubectl/pkg/cmd/config/rename_context.go:67]: Variables cannot be translated. I think the string inside this variable needs to be translated

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/config/rename_context.go:67]: Yeah, you should use `i18n.T` on the string in `renameContextShort` variable text.

File: staging/src/k8s.io/kubectl/pkg/cmd/config/rename_context.go [hunk 9]
```diff
@@ -20,6 +20,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/config/rename_context.go:23]: This import should be with the rest of k8s imports.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go [hunk 10]
```diff
@@ -19,6 +19,7 @@ package create
 import (
 	"context"
 	"fmt"
+	"k8s.io/kubectl/pkg/util/i18n"
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go:22]: Move down to the rest of k8s imports.

File: staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout.go [hunk 11]
```diff
@@ -27,8 +27,8 @@ import (
 )
 
 var (
-	rolloutLong = templates.LongDesc(`
-		Manage the rollout of a resource.` + rolloutValidResources)
+	rolloutLong = templates.LongDesc(i18n.T(`
+		Manage the rollout of a resource.` + rolloutValidResources))
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout.go:31]: `rolloutValidResources` should not be inside `i18n.T` invocation.

File: staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go (hunk 7 above)
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go:46]: Fair point, not needed.

**Review by soltysh — CHANGES_REQUESTED** [review]
There are still a few comments. Also please squash all your changes into a single commit.

**Comment by HerrNaN** [comment]
@soltysh @ramnar Your comments should be resolved now!

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/triage accepted
/priority backlog

**Pull request merged**